internal/prefixcollector/kubeadm_prefix_source.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prefixcollector
+
+import (
+	"cmd-exclude-prefixes-k8s/internal/utils"
+	"context"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	apiV1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/apimachinery/pkg/watch"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2"
+)
+
+const (
+	// KubeNamespace is KubeAdm ConfigMap namespace
+	KubeNamespace = "kube-system"
+	// kubeName is KubeAdm ConfigMap name
+	kubeName   = "kubeadm-config"
+	bufferSize = 4096
+)
+
+// KubeAdmPrefixSource is KubeAdm ConfigMap excluded prefix source
+type KubeAdmPrefixSource struct {
+	configMapInterface v1.ConfigMapInterface
+	prefixes           *utils.SynchronizedPrefixesContainer
+	ctx                context.Context
+	notify             Notifier
+}
+
+// Prefixes returns prefixes from source
+func (kaps *KubeAdmPrefixSource) Prefixes() []string {
+	return kaps.prefixes.Load()
+}
+
+// NewKubeAdmPrefixSource creates KubeAdmPrefixSource
+func NewKubeAdmPrefixSource(ctx context.Context, notify Notifier) *KubeAdmPrefixSource {
+	clientSet := KubernetesInterface(ctx)
+	configMapInterface := clientSet.CoreV1().ConfigMaps(KubeNamespace)
+	kaps := KubeAdmPrefixSource{
+		configMapInterface: configMapInterface,
+		ctx:                ctx,
+		notify:             notify,
+		prefixes:           utils.NewSynchronizedPrefixesContainer(),
+	}
+
+	go kaps.watchKubeAdmConfigMap()
+	return &kaps
+}
+
+func (kaps *KubeAdmPrefixSource) watchKubeAdmConfigMap() {
+	kaps.checkCurrentConfigMap()
+	configMapWatch, err := kaps.configMapInterface.Watch(kaps.ctx, metav1.ListOptions{})
+	if err != nil {
+		logrus.Errorf("Error creating config map watch: %v", err)
+		return
+	}
+
+	for kaps.ctx.Err() == nil {
```

Comment:
Now we can remove this check 
```suggestion
	for {
```
and use  `case <-kaps.ctx.Done():` as a first case in the statement `select`. Related to all similar places.

Reply:
Fixed.